### PR TITLE
[APPS-SD] Fix a few bugs and bring it up to speed with SD CLI

### PR DESCRIPTION
**Pull request by Abhishek-Varma**
-- This commit fixes a few bugs in `apps`'s SD CLI and brings it
   up to speed with SD CLI.

Signed-off-by: Abhishek Varma <abhishek@nod-labs.com>

### CODE REVIEW

**Review by powderluv — APPROVED** [review]
Please check the webui works too. @Shukla-Gaurav  please take a look. I'll rebase my patch on this

File: apps/stable_diffusion/src/models/opt_params.py [hunk 1]
```diff
@@ -13,6 +16,14 @@
...
 }
 
+# TODO: Implement a better way to handle the KeyNotFound error by delaying the invocation.
+variant = None
+version = None
+try:
+    variant, version = hf_model_variant_map[args.hf_model_id]
+except:
+    pass
+
```
Comment (Shukla-Gaurav) [inline apps/stable_diffusion/src/models/opt_params.py:26]: This has to be local to each function, i.e., means inside get_unet(), get_vae() and get_clip(), as these functions can be called multiple times with different model ids.

File: apps/stable_diffusion/src/models/opt_params.py [hunk 2]
```diff
@@ -60,7 +71,6 @@ def get_params(bucket_key, model_key, model, is_tuned, precision):
 
 
 def get_unet():
-    variant, version = hf_model_variant_map[args.hf_model_id]
```
Comment (Shukla-Gaurav) [inline apps/stable_diffusion/src/models/opt_params.py:63]: This is necessary to not fix the version and variant when executing it multiple times.

File: apps/stable_diffusion/src/models/__init__.py [hunk 3]
```diff
@@ -1,2 +1,10 @@
 from .model_wrappers import SharkifyStableDiffusionModel
-from .opt_params import get_vae, get_unet, get_clip, get_tokenizer
+from .opt_params import (
+    get_vae,
+    get_unet,
+    get_clip,
+    get_tokenizer,
+    get_params,
+    version,
+    variant,
```
Comment (Shukla-Gaurav) [inline apps/stable_diffusion/src/models/__init__.py:9]: We should avoid such global variables(as they change with multiple calls) and importing them. As with web, there will be multiple calls with changes to model id's.

File: apps/stable_diffusion/src/utils/sd_annotation.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+
+device = (
+    args.device if "://" not in args.device else args.device.split("://")[0]
+)
+
+
+# Download the model (Unet or VAE fp16) from shark_tank
+def load_model_from_tank():
+    from ..models import get_params, version, variant
```
Comment (Shukla-Gaurav) [inline apps/stable_diffusion/src/utils/sd_annotation.py:20]: Please fix relative imports. They have problem with pyinstaller.

File: apps/stable_diffusion/src/utils/sd_annotation.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    download_public_file,
+    WORKDIR,
+)
+from shark.parser import shark_args
+from .stable_args import args
+
+
+device = (
+    args.device if "://" not in args.device else args.device.split("://")[0]
+)
```
Comment (Shukla-Gaurav) [inline apps/stable_diffusion/src/utils/sd_annotation.py:15]: avoid such global variables.

**Review by Shukla-Gaurav — CHANGES_REQUESTED** [review]
Thanks, needs following changes.

File: apps/stable_diffusion/src/utils/sd_annotation.py (hunk 5 above)
Comment (dan-garvey) [reply apps/stable_diffusion/src/utils/sd_annotation.py:15]: agree here and elsewhere with what gaurav said. We want to move away from global variables and dependence on args outside of the script that is called from the command line

**Review by Shukla-Gaurav — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**